crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -572,6 +572,90 @@ impl Mesh {
         Ok(self)
     }
 
+    /// Merges the [`Mesh`] data of `other` with `self`. The attributes and indices of `other` will be appended to `self`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the vertex attribute values of `other` are incompatible with `self`.
+    /// For example, [`VertexAttributeValues::Float32`] is incompatible with [`VertexAttributeValues::Float32x3`].
+    pub fn merge(&mut self, other: Mesh) {
+        use VertexAttributeValues::*;
+
+        // The indices of `other` should start after the last vertex of `self`.
+        let index_offset = self.count_vertices();
```

Comment:
Note that (from `count_vertices`'s documentation):

> If the attributes have different vertex counts, the smallest is returned.

So it may be silently wrong (for your usecase) if that's the case. Might also be something to document.

Reply:
I changed it to just use the vertex position attribute length for now, although I'm not entirely sure if that's fully correct either